File: app/Http/Controllers/SpecialistPatternsJava/Creational/AbstractFactory/AbstractFactoryJava/src/windows/PlasticFrameWindow.java
package windows;

import static java.lang.System.out;

// Класс PlasticFrameWindow представляет окно с пластиковой рамой.
public class PlasticFrameWindow implements Window {

    // Метод open() открывает окно.
    @Override
    public void open() {
        out.println("Открытие пластикового окна");
    }

    // Метод close() закрывает окно.
    @Override
    public void close() {
        out.println("Закрытие пластикового окна");
    }

    // Метод install() выполняет установку окна.
    @Override
    public Window install() {
        out.println("Установка пластикового окна");
        return this;
    }
}

File: app/Http/Controllers/SpecialistPatterns/AbstractFactory/Windows/PlasticFrameWindow.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\AbstractFactory\Windows;


// Класс PlasticFrameWindow представляет окно с пластиковой рамой.
use Illuminate\Support\Facades\Log;

class PlasticFrameWindow implements Window {

    // Метод open() открывает окно.
    public function open() {
        Log::channel('sergey')->info("Открытие пластикового окна");
    }

    // Метод close() закрывает окно.
    public function close() {
        Log::channel('sergey')->info("Закрытие пластикового окна");
    }

    // Метод install() выполняет установку окна.
    public function install(): Window {
        Log::channel('sergey')->info("Установка пластикового окна");
        return $this;
    }
}
